contracts/voting/dao-voting-native-staked/src/contract.rs

```diff
@@ -1,24 +1,39 @@
 #[cfg(not(feature = "library"))]
 use cosmwasm_std::entry_point;
+
 use cosmwasm_std::{
-    coins, to_binary, BankMsg, Binary, CosmosMsg, Deps, DepsMut, Env, MessageInfo, Response,
-    StdResult, Uint128,
+    coins, to_binary, BankMsg, BankQuery, Binary, CosmosMsg, Decimal, Deps, DepsMut, Env,
+    MessageInfo, Reply, Response, StdResult, SubMsg, Uint128, Uint256, WasmMsg,
 };
 use cw2::set_contract_version;
 use cw_controllers::ClaimsResponse;
-use cw_utils::{must_pay, Duration};
+use cw_utils::{must_pay, parse_reply_instantiate_data, Duration};
 use dao_interface::state::Admin;
-use dao_interface::voting::{TotalPowerAtHeightResponse, VotingPowerAtHeightResponse};
+use dao_interface::voting::{
+    IsActiveResponse, TotalPowerAtHeightResponse, VotingPowerAtHeightResponse,
+};
+use dao_voting::threshold::ActiveThreshold;
 
 use crate::error::ContractError;
+use crate::hooks::{stake_hook_msgs, unstake_hook_msgs};
 use crate::msg::{
-    ExecuteMsg, InstantiateMsg, ListStakersResponse, MigrateMsg, QueryMsg, StakerBalanceResponse,
+    ActiveThresholdResponse, DenomResponse, ExecuteMsg, GetHooksResponse, InstantiateMsg,
+    ListStakersResponse, MigrateMsg, QueryMsg, StakerBalanceResponse, TokenInfo,
+};
+use crate::state::{
+    Config, ACTIVE_THRESHOLD, CLAIMS, CONFIG, DAO, DENOM, HOOKS, MAX_CLAIMS, STAKED_BALANCES,
+    STAKED_TOTAL,
 };
-use crate::state::{Config, CLAIMS, CONFIG, DAO, MAX_CLAIMS, STAKED_BALANCES, STAKED_TOTAL};
 
 pub(crate) const CONTRACT_NAME: &str = "crates.io:dao-voting-native-staked";
 pub(crate) const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
 
+const INSTANTIATE_TOKEN_REPLY_ID: u64 = 0;
+
+// We multiply by this when calculating needed power for being active
+// when using active threshold with percent
+const PRECISION_FACTOR: u128 = 10u128.pow(9);
```

Comment:
should this be configurable?

Reply:
no i think this is fancy math @0xekez worked out